### PR TITLE
Refine comment editing

### PR DESCRIPTION
**Pull request by JoshSmith**
Had to reopen this due to weird rebase issues.

I have broken out issues #129, #130, and #131 to not continue to make this PR larger than it needs to be.

I did sneak in some things like image optimizations and refactoring of CSS. And I'm sure I touched a number of other places.

Everything I wanted to test appears to be tested. Now could just use a review.

### CODE REVIEW

File: app/templates/components/code-theme-selector.hbs [hunk 1]
```diff
@@ -0,0 +1 @@
+{{yield}}
```
Comment (begedin) [inline app/templates/components/code-theme-selector.hbs]: Do we actually have a reason to yield here? As in, do we expect to use the component in a way that provides it HTML to render? If not, we could just have an empty file here instead (and remove the test that ensures yielding works, if there is one).

Comment (JoshSmith) [inline app/templates/components/code-theme-selector.hbs]: I don't believe so. Not sure there is such a test either.

File: app/templates/project/posts/post.hbs [hunk 2]
```diff
@@ -2,10 +2,12 @@
 <div class="post-timeline">
   {{post-details post=post}}
   {{post-comment-list comments=comments}}
+  <div>
   {{create-comment-form comment=newComment saveComment='saveComment'}}
   {{#if error}}
     {{error-formatter error=error}}
   {{/if}}
+  </div>
```
Comment (begedin) [inline app/templates/project/posts/post.hbs]: Missing some indentation between these div tags. Unless we prefer not to have it here?

Comment (JoshSmith) [inline app/templates/project/posts/post.hbs]: Just an oversight. Can indent.

File: app/templates/components/code-theme-selector.hbs (hunk 1 above)
Comment (begedin) [inline app/templates/components/code-theme-selector.hbs]: It's usually auto-generated if you're generating a component, so there might be.

Comment (JoshSmith) [inline app/templates/components/code-theme-selector.hbs]: There's a component test, but I think I modified it.

**Comment by begedin** [comment]
Looks good, other than the couple of very minor details. Feel free to merge :+1:

**Pull request merged**